src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,196 @@
+        }
+        return List.of(keys);
+    }
+
+    @Override
+    public List<V> getValues() {
+        V[] keys = (V[]) new Object[size];
+        for (int i = 0, j = 0; i < table.length; i++) {
+            Node<K, V> bucket = table[i];
+            while (bucket != null) {
+                keys[j++] = bucket.value;
+                bucket = bucket.next;
+            }
+        }
+        return List.of(keys);
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static int hash(Object key) {
+        return key == null ? 0 : key.hashCode();
+    }
+
+    private int getBucketIndex(int hash) {
+        return table.length - 1 & hash;
+    }
+
+    /**
+     * Because we are using power-of-two expansion,
+     * the elements from each bucket must either stay at same index,
+     * or move with a power of two offset in the new table.
+     */
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[table.length << 1];
+        threshold = (int) (table.length * LOAD_FACTOR);
+        for (int i = 0; i < oldTable.length; i++) {
+            Node<K, V> bucket = oldTable[i];
+            if (bucket != null) {
+                if (bucket.next == null) {
+                    table[getBucketIndex(bucket.hash)] = bucket;
+                } else {
+                    Node<K, V> sameIndexBucketHead = null;
+                    Node<K, V> sameIndexBucketTail = null;
+                    Node<K, V> offsetBucketHead = null;
+                    Node<K, V> offsetBucketTail = null;
+                    while (bucket != null) {
+                        if ((bucket.hash & oldTable.length) == 0) {
+                            if (sameIndexBucketTail == null) {
+                                sameIndexBucketHead = bucket;
+                            } else {
+                                sameIndexBucketTail.next = bucket;
+                            }
+                            sameIndexBucketTail = bucket;
+                        } else {
+                            if (offsetBucketTail == null) {
+                                offsetBucketHead = bucket;
+                            } else {
+                                offsetBucketTail.next = bucket;
+                            }
+                            offsetBucketTail = bucket;
+                        }
+                        bucket = bucket.next;
+                    }
+                    if (sameIndexBucketTail != null) {
+                        sameIndexBucketTail.next = null;
+                        table[i] = sameIndexBucketHead;
+                    }
+                    if (offsetBucketTail != null) {
+                        offsetBucketTail.next = null;
+                        table[i + oldTable.length] = offsetBucketHead;
+                    }
+                }
+            }
```

Comment:
Instead of all this code just go through all nodes from old table and call your put() method for each of them. As you already have new main table with new capacity then this method will determine automatically to which bucket number node should go

Reply:
Done, but i suppose it would create many-many other objects and slow down MyHash performance